tools/fix_go_deps.sh

```diff
@@ -5,12 +5,17 @@ set -euo pipefail
 : "${GAZELLE_PATH:=}"
 : "${BAZEL_COMMAND:=bazelisk}"
 
-GAZELLE_COMMAND=("$BAZEL_COMMAND" run --build_metadata=DISABLE_COMMIT_STATUS_REPORTING=true //:gazelle --config=buildbuddy_bes_backend --config=buildbuddy_bes_results_url --)
+CI_FLAGS=""
+if [[ "$CI" == true ]]; then
+  CI_FLAGS=(--config=buildbuddy_bes_backend --config=buildbuddy_bes_results_url)
+fi
+
+GAZELLE_COMMAND=("$BAZEL_COMMAND" run --build_metadata=DISABLE_COMMIT_STATUS_REPORTING=true //:gazelle "${CI_FLAGS[@]}" --)
 if [[ "$GAZELLE_PATH" ]]; then
   GAZELLE_COMMAND=("$GAZELLE_PATH")
 fi
 
-GO_COMMAND=("$BAZEL_COMMAND" run --build_metadata=DISABLE_COMMIT_STATUS_REPORTING=true //:go --config=buildbuddy_bes_results_url --config=buildbuddy_bes_backend --)
+GO_COMMAND=("$BAZEL_COMMAND" run --build_metadata=DISABLE_COMMIT_STATUS_REPORTING=true //:go "${CI_FLAGS[@]}" --)
```

Comment:
nit: put bazel flags before the target for consistency (or if you prefer putting them after the target, could you update the existing flags?)

```suggestion
GO_COMMAND=("$BAZEL_COMMAND" run --build_metadata=DISABLE_COMMIT_STATUS_REPORTING=true "${CI_FLAGS[@]}" //:go --)
```